src/components/SearchBar/SearchBar.tsx

```diff
@@ -1,51 +1,84 @@
-import React from 'react';
+import React, { useRef } from 'react';
 
 import SearchIcon from '@assets/icons/search_glass.svg?react';
 
+import { useFilterContext } from '@/hooks/useFilterContext';
+import { useProductsDataContext } from '@/hooks/useProductsDataContext';
+import { SortByEnum } from '@/interfaces/Filters';
 import { CustomSelector } from '@/ui/CustomSelector/CustomSelector';
 import FilterButton from '@/ui/FilterButton/FilterButton';
 
 import styles from './SearchBar.module.css';
 
+const selectors = [
```

Comment:
use readonly type for this constant
so create a type for this selectros and make it readonly, this is for type-safe
https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype